Documentation test triggered without ```

If you'd like to fix this bug, [here's how!](https://github.com/rust-lang/rust/issues/49717#issuecomment-392547535)

------------------------

The follow documentation comment is considered a documentation test.  All of the Rust references I have read indicate ``` are needed to indicate a documentation test.

## lib.rs

    ///                   
    ///     a             
    /// b                 
    pub fn some_lib_fn() {
                      
    } 

Then running `cargo test` gives the following output:

     running 0 tests                                                                   
                                                                                       
     test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out        
                                                                                       
        Doc-tests reproduce-doc-test                                                   
                                                                                       
     running 1 test                                                                    
     test src/lib.rs - some_lib_fn (line 1) ... FAILED                                 
                                                                                       
     failures:                                                                         
                                                                                       
     ---- src/lib.rs - some_lib_fn (line 1) stdout ----                                
             error[E0425]: cannot find value `a` in this scope                         
      --> src/lib.rs:2:1                                                               
       |                                                                               
     3 | a                                                                             
       | ^ not found in this scope                                                     
                                                                                       
     thread 'rustc' panicked at 'couldn't compile the test', librustdoc/test.rs:298:13 
     note: Run with `RUST_BACKTRACE=1` for a backtrace.                                
                                                                                       
                                                                                       
     failures:                                                                         
         src/lib.rs - some_lib_fn (line 1)                                             
                                                                                       
     test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out    
                                                                                       
     error: test failed, to rerun pass '--doc'

--- a/src/doc/rustdoc/src/documentation-tests.md
+++ b/src/doc/rustdoc/src/documentation-tests.md
@@ -284,3 +284,27 @@ environment that has no network access.
 compiles, then the test will fail. However please note that code failing
 with the current Rust release may work in a future release, as new features
 are added.
+
+## Syntax reference
+
+The *exact* syntax for code blocks, including the edge cases, can be found
+in the [Fenced Code Blocks](https://spec.commonmark.org/0.28/#fenced-code-blocks)
+section of the CommonMark specification.
+
+Rustdoc also accepts *indented* code blocks as an alternative to fenced
+code blocks: instead of surrounding your code with three backticks, you
+can indent each line by four or more spaces.
+
+``````markdown
+    let foo = "foo";
+    assert_eq!(foo, "foo");
+``````
+
+These, too, are documented in the CommonMark specification, in the
+[Indented Code Blocks](https://spec.commonmark.org/0.28/#indented-code-blocks)
+section.
+
+However, it's preferable to use fenced code blocks over indented code blocks.
+Not only are fenced code blocks the more popular choice in the Rust community,
+but there is no way to use directives such as `ignore` or `should_panic` with
+indented code blocks.

--- a/src/doc/rustdoc/src/documentation-tests.md
+++ b/src/doc/rustdoc/src/documentation-tests.md
@@ -305,6 +305,6 @@ These, too, are documented in the CommonMark specification, in the
 section.
 
 However, it's preferable to use fenced code blocks over indented code blocks.
-Not only are fenced code blocks the more popular choice in the Rust community,
+Not only are fenced code blocks considered more idiomatic for Rust code,
 but there is no way to use directives such as `ignore` or `should_panic` with
 indented code blocks.

